### PR TITLE
Fix ratio plot for uGT data-emulator mismatches

### PR DESCRIPTION
**Pull request by dinyar**
#### PR description:

Recently it was noticed that the uGT data-emulator mismatch ratio plot could show more than 100% mismatches. The cause for this is that while the denominator is filled once per event, the numerator is filled for each mismatching algorithm bit.

This PR changes the logic such that the numerator is filled only once per event. Arguably it would be nicer to instead fill the denominator with all valid algo bits, however to my knowledge it's not trivial to retrieve the number of algorithms used by a given menu from the DQM.

#### PR validation:

None yet.

### CODE REVIEW

File: DQM/L1TMonitor/src/L1TdeStage2uGT.cc [hunk 1]
```diff
@@ -157,21 +164,35 @@ void L1TdeStage2uGT::analyze(const edm::Event& event, const edm::EventSetup& es)
...
               hname = "EmulatorNoData_" + bxt.str();
-              fillHist(m_SummaryHistograms, hsummary, float(NFinalMismatchEmulNoData), 1.);
+              foundFinalMismatchEmulNoData = true;
               wt = -1;
             }
             fillHist(m_HistNamesFinal, hname, float(algoBit), 1.);
-            finalDecisionMismatches_vs_LS->Fill(float(lumi), wt);
+            finalDecisionMismatches_vs_LS->Fill(
+                float(lumi),
+                wt);  // TODO: I think this weight makes things confusing. If we have uncorrelated no emu and a few no data mismatches it would look like we have fewer mismatches than we actually do.
```
Comment (dinyar) [inline DQM/L1TMonitor/src/L1TdeStage2uGT.cc:177]: I spotted this (and the same line further below) while editing the file. If I understand correctly we're incrementing that histogram when we have a data, but no emulator mismatch and decrementing it when we have a emulator but no data mismatch. I think this doesn't make much sense, we should probably have two histograms in my opinion..

**Comment by tjavaid** [comment]
please test

File: DQM/L1TMonitor/src/L1TdeStage2uGT.cc (hunk 1 above)
Comment (dinyar) [reply DQM/L1TMonitor/src/L1TdeStage2uGT.cc]: I split the vs. LS histograms in two as proposed above.

**Comment by tjavaid** [comment]
please test

**Comment by dinyar** [comment]
Hi @tjavaid, @syuvivida,

From my side this is ready. This isn't super critical for the next few days of running, so no stress..

Also pinging @aloeliger, @epalencia, and @vukasinmilosevic from the L1T side just for information.

Cheers,
Dinyar

**Comment by smuzaffar** [comment]
Milestone for this pull request has been moved to CMSSW_14_0_X.Please open a backport if it should also go in to CMSSW_13_3_X.

**Comment by tjavaid** [comment]
please test

**Comment by dinyar** [comment]
Hi @tjavaid,

I checked the DQM bin by bin comparison and this looks good. In particular for wf140.56 one can see that the ratio plot of the uGT is much better behaved than before.

Cheers,
Dinyar

**Comment by antoniovilela** [comment]
+1

**Pull request merged**